### PR TITLE
feat: Search bar on homepage

### PR DESCRIPTION
**Pull request by stanleyjones**
**Placeholder**
Screenshot 2023-03-23 at 11 54 20 AM

**Validates as you type**
Screenshot 2023-03-23 at 11 54 42 AM

**Works for block height**
Screenshot 2023-03-23 at 11 54 54 AM

**Works for transaction hashes**
Screenshot 2023-03-23 at 11 55 08 AM

**Works for addresses, including subresources**
Screenshot 2023-03-23 at 11 55 24 AM

### CODE REVIEW

File: client/src/ui/search.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  InputRightElement,
+  SearchIcon,
+} from "@liftedinit/ui";
+import { FormEvent, useState } from "react";
+import { NavigateFunction, useNavigate } from "react-router-dom";
+
+const validTerms = [
+  { path: "blocks", regex: /^\d+$/ },
+  { path: "transactions", regex: /^[0-9a-f]{64}$/ },
+  { path: "address", regex: /^m[0-9a-zA-Z]{49,54}$/ },
```
Comment (stanleyjones) [inline client/src/ui/search.tsx:15]: Would love your eyes on these, @hansl . We could obviously get more fancy (block height has to be less than current height, etc.) but does anything jump out as totally wrong?

**Comment by tysloan** [comment]
@stanleyjones do we plan for the search bar to move to the header when a user navigates to another page?

**Comment by stanleyjones** [comment]
> @stanleyjones do we plan for the search bar to move to the header when a user navigates to another page?

Oops. I now see that's in the issue description. Hmm... Let me take a stab at that. The header isn't super accommodating at the moment.

**Comment by stanleyjones** [comment]
Updated...

Screenshot 2023-03-24 at 10 33 17 AM

**Pull request merged**